### ISSUE TITLE
buildin function merge got deprecated in puppet 8

### ISSUE THREAD

**Issue by suchAdemon** [voxpupuli/puppet-unattended_upgrades#228]
## Affected Puppet, Ruby, OS and module versions/distributions

- PuppetServer: 8.0.0
- PuppetAgent: 8.0.0
- Ruby: 3.1.2p20
- Distribution: Debuan 12
- Module version:  7.1.0+ (commit 69eb8b724f97b418ef747bcb02e3ee55f8fb9501)

## How to reproduce (e.g Puppet code you use)

Just perform a run with puppet 8

## What are you seeing

```
Error: Could not retrieve catalog from remote server: Error 500 on SERVER: Server Error: Evaluation Error: Error while evaluating a Resource Statement, Evaluation Error: Error while evaluating a Function Call, deprecation. merge. This function is deprecated, please use stdlib::merge instead. at ["<..>/modules/unattended_upgrades/manifests/init.pp", 35] (file: "<..>/modules/unattended_upgrades/manifests/init.pp, line: 35, column: 11) on node <..>
```

## Possible solution
> using puppet internal function

```git
diff --git a/manifests/init.pp b/manifests/init.pp
index a4e4b98..5b7af1e 100644
--- a/manifests/init.pp
+++ b/manifests/init.pp
@@ -32,19 +32,19 @@ class unattended_upgrades (
   # apt::conf settings require the apt class to work
   include apt
 
-  $_age = merge($unattended_upgrades::default_age, $age)
+  $_age = $unattended_upgrades::default_age + $age
   assert_type(Unattended_upgrades::Age, $_age)
 
-  $_auto = merge($unattended_upgrades::default_auto, $auto)
+  $_auto = $unattended_upgrades::default_auto + $auto
   assert_type(Unattended_upgrades::Auto, $_auto)
 
-  $_backup = merge($unattended_upgrades::default_backup, $backup)
+  $_backup = $unattended_upgrades::default_backup + $backup
   assert_type(Unattended_upgrades::Backup, $_backup)
 
-  $_mail = merge($unattended_upgrades::default_mail, $mail)
+  $_mail = $unattended_upgrades::default_mail + $mail
   assert_type(Unattended_upgrades::Mail, $_mail)
 
-  $_upgradeable_packages = merge($unattended_upgrades::default_upgradeable_packages, $upgradeable_packages)
+  $_upgradeable_packages = $unattended_upgrades::default_upgradeable_packages + $upgradeable_packages
   assert_type(Unattended_upgrades::Upgradeable_packages, $_upgradeable_packages)
 
   package { 'unattended-upgrades':
```

**Comment by suchAdemon** [comment]
Hello,
yea true merge is not the build in function, was still thinking about the code snippet I posted, if it could break something. I guess then there will be no interim commit to sort it out, so that the main branch could continue updates and being used on the latest main commit/or similar?
If not, then I will keep my local changes till the long-term fix got merged.

Btw, cool thing that you started to namespace functions.
